### DISCUSSION TITLE
Https/Websocket

### DISCUSSION THREAD

**Discussion by jbbcst** [cesanta/mongoose#1780] [Ask questions, get answers] [answered]
Hi,
I have a server (based on mongoose HTTP server example using TLS) that listening on https://0.0.0.0:443, is it possible to add a websocket management with the same mg_http_listen or should I create a second one? 
I look at  the websocket-server example and it listens on ws://localhost:8000.

**Comment by scaprile** [comment, marked as answer]
A WS connection upgrades from an HTTP connection.
You can have as many listeners in as many ports as your memory allows.
In every listener you can have as many URIs as you want and upgrade those to WS, it doesn't have to be a separate port; just check any of the many other examples and tutorials like for example the device dashboard or the json-rpc processor. http://mongoose.ws/tutorials

In other words, you can upgrade to WS on https://0.0.0.0:443/websocket or https://0.0.0.0:443/someuri or https://0.0.0.0:8000 or https://0.0.0.0:someport or https://0.0.0.0:someport/someuri. The choice is yours. You just need some way to know where to upgrade to WS, be it a different port or some unique URI

**Reply by jbbcst** [reply]
ok thanks
